### PR TITLE
Add basic documentation to the readme

### PR DESCRIPTION
**Pull request by ahuth**
Perhaps a replacement for #7, this PR adds basic documentation about the project to the README. It's not intended to be comprehensive, and I'll be adding more documentation of providers and consumers.

@sharmilajesupaul can you take a look at this, please?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,43 @@
...
+        render: () => <Button disabled>Nope</Button>
+      }
+    ],
+  };
+}
+```
+
+We also have "consumers" that use the variation provider to perform tasks. Example consumers are:
+* Rendering variations in [Storybook](https://storybook.js.org/)
+* Running in tests with [Enzyme](https://airbnb.io/enzyme/)
```
Comment (ljharb) [inline README.md]: we should link to https://github.com/airbnb/react-component-variations-consumer-enzyme here

Comment (ahuth) [reply README.md]: Good call

File: README.md [hunk 2]
```diff
@@ -2,6 +2,43 @@
...
+    ],
+  };
+}
+```
+
+We also have "consumers" that use the variation provider to perform tasks. Example consumers are:
+* Rendering variations in [Storybook](https://storybook.js.org/)
+* Running in tests with [Enzyme](https://github.com/airbnb/react-component-variations-consumer-enzyme)
+* Taking screenshots with [Happo](https://happo.io/)
+* Checking for accessibility violations with [Axe](https://www.deque.com/axe/)
```
Comment (lencioni) [inline README.md]: Should these link to the consumers as well?

**Comment by ahuth** [comment]
Also /cc @indiesquidge

File: README.md (hunk 2 above)
Comment (ahuth) [reply README.md]: Yeah, definitely. The axe one isn't open-source, yet, though

**Pull request merged**